cluster-controller/src/main/java/io/strimzi/controller/cluster/resources/Labels.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.controller.cluster.resources;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonMap;
+import static java.util.Collections.unmodifiableMap;
+
+/**
+ * An immutable set of labels
+ */
+public class Labels {
+
+    public static final String STRIMZI_DOMAIN = "strimzi.io/";
+
+    /**
+     * The kind of a ConfigMap:
+     * <ul>
+     *     <li>{@code strimzi.io/kind=cluster}
+     *         identifies a ConfigMap that is intended to be consumed by
+     *         the cluster controller.</li>
+     *     <li>{@code strimzi.io/kind=topic}
+     *         identifies a ConfigMap that is intended to be consumed
+     *         by the topic controller.</li>
+     * </ul>
+     */
+    public static final String STRIMZI_KIND_LABEL = STRIMZI_DOMAIN + "kind";
+    /**
+     * The type of Strimzi component:
+     * E.g: {@code kafka}, {@code zookeeper}, {@code topic-controller},
+     *      {@code connect}, {@code connect-s2i}
+     */
+    public static final String STRIMZI_TYPE_LABEL = STRIMZI_DOMAIN + "type";
+
+    /**
+     * The Strimzi cluster the resource is part of.
+     * The value is the cluster name (i.e. the name of the cluster CM)
+     */
+    public static final String STRIMZI_CLUSTER_LABEL = STRIMZI_DOMAIN + "cluster";
+
+    /**
+     * The name of the K8S resource.
```

Comment:
Unrelated ... but I think that thi things like ConfigMaps, StatefulSets etc. are Kubernetes resources. OpenShift resources are things like ImageStream, DeploymentConfig etc.

I'm personally fine with k8s in javadoc and comments in code. The real docu should use Kubernetes only.

Reply:
Maybe we don't want to use Kubernetes/OpenShift but using acronym in the doc isn't so good fmpov. Saying Kubernetes is better then k8s. Of course k8s is good for source code (i.e. var names ...)

Reply:
Since this isn't a library or framework I think we need to be honest about who will be reading these javadocs: Strimzi developers and more-or-less no one else. Therefore I think more informal javadoc are acceptable. I agree with @scholzj that for what's in the `documentation/` directory we should be less informal.

Reply:
Exactly. This is not a documentation. This is a javadoc of an internal API.

Reply:
I continue to not agree but feel free to ignore my comment ;)